crates/solver/src/settlement.rs

```diff
@@ -1596,7 +1596,7 @@ pub mod tests {
 
             assert_eq!(
                 some_surplus.total_surplus(&external_prices).to_integer(),
-                BigInt::from(1000_u128),
+                BigInt::from(500_u128),
```

Comment:
Surplus was 1000 here but now we take 50% as protocol fee so the surplus becomes 500.